<launch>
    <arg name="use_sim_time" default="true"/>
    <arg name="log_level" default="warn"/>

    <arg name="src_image" default="/sensing/camera/traffic_light/image_raw/compressed"/>
    <arg name="src_info" default="/sensing/camera/traffic_light/camera_info"/>
    <arg name="resized_image" default="/sensing/camera/undistorted/image_raw"/>
    <arg name="resized_info" default="/sensing/camera/undistorted/camera_info"/>

    <!-- ========================================= -->
    <!-- ============ undistort & resize ========= -->
    <!-- ========================================= -->
    <node name="undistort" pkg="vector_map_visual_localizer" exec="undistort_node" output="screen" args="--ros-args --log-level $(var log_level)">
        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <param name="width" value="800"/>
    </node>

    <!-- ========================================= -->
    <!-- ========== line segment detector ======== -->
    <!-- ========================================= -->
    <node if="false" name="lsd" pkg="vector_map_visual_localizer" exec="lsd_node" output="screen" args="--ros-args --log-level $(var log_level)">
        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <param name="max_range" value="$(var max_range)"/>
        <param name="image_size" value="$(var image_size)"/>
        <param name="line_thick" value="2"/>
        <param name="line_threshold" value="0.01"/>
        <param name="dilate_size" value="1"/>

        <remap from="$(var src_image)" to="$(var resized_image)"/>
        <remap from="$(var src_info)" to="$(var resized_info)"/>
    </node>

    <!-- ========================================= -->
    <!-- ========= vanish point estimator ======== -->
    <!-- ========================================= -->
    <node if="true" name="vanish" pkg="vector_map_visual_localizer" exec="vanish_node" output="screen" args="--ros-args --log-level $(var log_level)">
        <remap from="$(var src_image)" to="$(var resized_image)"/>
        <remap from="$(var src_info)" to="$(var resized_info)"/>
    </node>

</launch>
